### ISSUE TITLE
Questions about the discrepancy between reported fine-tuning results and my own test results on fine-tuning

### ISSUE THREAD

**Issue by yahyunee** [wjq-learning/CBraMod#6]
Hi,

I’ve fine-tuned on three different datasets (FACED, SEED-V, and PhysioNet-MI) and obtained results that differ significantly from those reported in your paper. I suspect I might have made a mistake but am unsure where to correct it.

I ran experiments with five different seed values (0, 1, 2, 3, 4) and averaged the results. I’ve attached an image summarizing the outcomes.

(Apologies for cropping parts of the tables from your paper.)
<img width="965" alt="Image" src="https://github.com/user-attachments/assets/37d76f94-d78d-4169-a221-52bfa4b2be5e" />
<img width="970" alt="Image" src="https://github.com/user-attachments/assets/c4e189f9-d15b-4fd4-b1cf-4d5e5c22850d" />

Additionally, I’ve attached the log files. Could you review them to identify any potential mistakes on my end? Also, if possible, could you share the seed values used in your experiments?

Thank you in advance for your help!

[log_0222_finetune_PhysioNet-MI_seed0.txt](https://github.com/user-attachments/files/18937713/log_0222_finetune_PhysioNet-MI_seed0.txt)
[log_0222_finetune_SEED-V_seed0.txt](https://github.com/user-attachments/files/18937715/log_0222_finetune_SEED-V_seed0.txt)
[log_0221_finetune_FACED_seed0.txt](https://github.com/user-attachments/files/18937714/log_0221_finetune_FACED_seed0.txt)

**Comment by wjq-learning** [comment]
Thank you for your feedback. I believe the results on the PhysioNet-MI dataset are very likely due to issues in data preprocessing. After carefully reviewing the code, I discovered a minor issue in the preprocessing code for the PhysioNet-MI dataset. I have already uploaded a corrected version, so please use the latest revised code to preprocess this dataset.

Regarding the results on the other two datasets, I believe they are not caused by bugs but are more likely influenced by random factors. A couple of days ago, I uploaded some code modifications to help reduce the impact of such randomness. For instance, I adjusted the activation function of the downstream classifier head, changing it from GELU to ELU. Additionally, I added a segmented learning rate setup to ensure that the pre-trained module and the downstream classifier head can be fine-tuned with different learning rates during downstream task training.

Please let me know if you encounter any further issues!

**Comment by yahyunee** [comment]
Hi, thanks for your quick reply! 
I've tried the replication on the newest version of your code, with random seed values set as (0,1,2,3,4) and got result as below. For FACED dataset I got higher accuracy while for SEED-V and PhysioNet-MI I got lower accuracy. Can you suggest adjustments that I can try out?

![Image](https://github.com/user-attachments/assets/f867a72f-c977-41a3-862e-d4c823ea9d04)

![Image](https://github.com/user-attachments/assets/f9e990b9-57af-4164-a109-dec6bc7b6b65)

I've attached the log files. Thank you for your support and assistance.

[log_0225_finetune_PhysioNet-MI_seed0.txt](https://github.com/user-attachments/files/18975636/log_0225_finetune_PhysioNet-MI_seed0.txt)
[log_0225_finetune_FACED_seed0.txt](https://github.com/user-attachments/files/18975634/log_0225_finetune_FACED_seed0.txt)
[log_0225_finetune_SEED-V_seed0.txt](https://github.com/user-attachments/files/18975635/log_0225_finetune_SEED-V_seed0.txt)

**Comment by Nullzyd** [comment]
> Hi,
> 
> I’ve fine-tuned on three different datasets (FACED, SEED-V, and PhysioNet-MI) and obtained results that differ significantly from those reported in your paper. I suspect I might have made a mistake but am unsure where to correct it.
> 
> I ran experiments with five different seed values (0, 1, 2, 3, 4) and averaged the results. I’ve attached an image summarizing the outcomes.
> 
> (Apologies for cropping parts of the tables from your paper.) <img alt="Image" width="965" src="https://private-user-images.githubusercontent.com/105848931/416094290-37d76f94-d78d-4169-a221-52bfa4b2be5e.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjkxNTYzODYsIm5iZiI6MTc2OTE1NjA4NiwicGF0aCI6Ii8xMDU4NDg5MzEvNDE2MDk0MjkwLTM3ZDc2Zjk0LWQ3OGQtNDE2OS1hMjIxLTUyYmZhNGIyYmU1ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMTIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDEyM1QwODE0NDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01MWQ1NWE5MTE2Zjc3NmU5MGNiNzk1N2MzMjI4YTgzNzQ5MDcyZDVhNTQ3MjdhZmQyNjQ4ZjA5ZTY0NmE1Mzg1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.hrpfdas-17MrekGNeDNdIZl8xOLuy8YUQqa9Yluej38"> <img alt="Image" width="970" src="https://private-user-images.githubusercontent.com/105848931/416094494-c4e189f9-d15b-4fd4-b1cf-4d5e5c22850d.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjkxNTYzODYsIm5iZiI6MTc2OTE1NjA4NiwicGF0aCI6Ii8xMDU4NDg5MzEvNDE2MDk0NDk0LWM0ZTE4OWY5LWQxNWItNGZkNC1iMWNmLTRkNWU1YzIyODUwZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMTIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDEyM1QwODE0NDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zYWQzMTg5MTE1ZDI3NTk1NjAzYWNmMGY0MTc1NzA4MzcyOGU1NGNjZjJjOTJkNTEwOTEyOTkzNTM1M2IwOGQxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.5Jg4FjVALn4Sin8hX0pTz4a_kwhs9H1wGZZibqgyoRQ">
> 
> Additionally, I’ve attached the log files. Could you review them to identify any potential mistakes on my end? Also, if possible, could you share the seed values used in your experiments?
> 
> Thank you in advance for your help!
> 
> [log_0222_finetune_PhysioNet-MI_seed0.txt](https://github.com/user-attachments/files/18937713/log_0222_finetune_PhysioNet-MI_seed0.txt) [log_0222_finetune_SEED-V_seed0.txt](https://github.com/user-attachments/files/18937715/log_0222_finetune_SEED-V_seed0.txt) [log_0221_finetune_FACED_seed0.txt](https://github.com/user-attachments/files/18937714/log_0221_finetune_FACED_seed0.txt)

我想问一下seed-v数据集和faced数据集在哪下载的，可以分享一下吗，谢谢
